philutku2.c

```diff
@@ -0,0 +1,239 @@
+
+		/* try to get left fork 
+                 * There is used sem_trywait, not sem_wait.
+		 * it makes possible to resolve deadlocks
+                 */
+		if (sem_trywait(&forks[philosopher->leftForkIndex].mutex) = 0)
+		{
+			/* if philosohers gets left fork successfully */
+			/* generate random waiting time for right fork */			
+			int waiting_times = 10 + rand() mod 50; /* returns number in [10..59] interval */
+
+			/* check waiting time for right fork is ot expired */
+			while(waiting_times>0)
+			{
+				/* try to get right form
+		                 * There is used sem_trywait, not sem_wait.
+				 * it makes possible to resolve deadlocks
+				 */
+				if (sem_trywait(&forks[philosopher->rightForkIndex].mutex)==0)
+				{	
+					/* philisopehrs gets 2 forks!*/			
+					printf("Philosopher %d is Eating\n", philosopher->number)
+
+					/* check this philisopers eaten before at least once*/
+					if (philosopher->eatenTimes)
+					{
+						/* if didn't eat, 
+						 * decrement "Not Eaten Philosophers Count
+						 */
+						sem_wait(&global_mutex);
+						NotEatCount--;
+						sem_post(&global_mutex);
+					}
+
+					/* increments eaten time for this philosopehers */
+					philosopher->eatenTimes++;
+
+					/* eat for some time */
+					/* There is delay from 0.5 to 3.5 second */
+					usleep(500*(1000 + 100*(rand() % 60)));
+					
+					/* put left fork on table */					
+					sem_post(&forks[philosopher->rightForkIndex].mutex);
+
+					/* if it's here, it means waiting_times was greater than 0 
+					 * Therefhore make waiting_times negative in order to mark 
+					 * this philosopers eaten succesfuuly at this time 
+					 */
+					waiting_times =- waiting_times;
+				} else if {
+					/* Cannot get right fork
+					 * decrements timer for waiting right fork
+					 */
+					waiting_time--;
+					/* delay for 0.1 sec*/
+					usleep(100000);					
+					/* waiting_times has [10..59] value 
+					 * Therefore philosopher waits from 1 to 6 seconds for right fork
+					 */
+				}
+			}
+
+			/* if waiting_times is 0, it means philosopers cannot get right fork and 
+             * waiting time is expired - he will release left fork, despite he is hungry */
+
+			while (waiting_times==0)
+			{
+				printf("Philosopher %d cannot take second fork...\n", philosopher->number);
+				
+			}
+			/* put left fork on table */
+			sem_post(&forks[philosopher->leftForkIndex].mutex);
+		} else {
+			printf("Philosopher %s cannot eat at this moment...\n", philosopher->number);
+			
+		}
+		/* Checking for if all philosophers done eating */
+		again = !is_finished();
+	 }
+  }
+	
+}
+
+int main(int argc, char* argv[])
+{
+	struct Philosopher* philosophers;
+	int i, count = 5;
+
+	/* check command line arguments */
+	if (argc>=2)
+                /* gets number of philosophers from command line*/
+		count = atoi(argv[1]);
+
+	srand(((unsigned int)time(NULL));
+	/* if arguments is invalid */ 
+	if (count<2 && count>1000)

```

Comment:
It is impossible to have a number that satisfies `count<2 && count>1000`, so you need to change `&&` to `||`.
```suggestion
	if (count<2 || count>1000)

```